OpenRA.Mods.Common/Activities/Move/Nudge.cs

```diff
@@ -0,0 +1,71 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class Nudge : Activity
+	{
+		readonly Mobile mobile;
+		readonly Aircraft aircraft;
+		readonly Actor nudger;
+		public Nudge(Actor self, Actor nudger)
+		{
+			this.nudger = nudger;
+
+			mobile = self.TraitOrDefault<Mobile>();
+			aircraft = self.TraitOrDefault<Aircraft>();
```

Comment:
Can we do something like
```suggestion
			var move = self.Trait<IMove>();
			mobile = move as Mobile;
			aircraft = move as Aircraft;
```
?

Reply:
sure, why not